pkg/pillar/cmd/nim/nim.go

```diff
@@ -791,7 +790,29 @@ func (n *nim) ingestDevicePortConfigFile(oldDirname string, newDirname string, n
 		return
 	}
 	dpc.DoSanitize(n.Log, true, false, "", true, true)
-	dpc.OriginFile = filename
+	if filename != "" {
+		// Determine ingested filename - we ignore anything but usb.json to match
```

Comment:
Actually, usb.json (as coming from a specially formatted USB stick) is being taken care of by device-steps.sh (it copies it directly to /run/global//DevicePortConfig).
In case of DPC being baked into an EVE image, customers usually rename override.template to override.json. So that's the filename we are more likely going to find in the /config partition.

Reply:
Maybe we should accept here both usb.json and override.json, just in case.

Reply:
Looking at this more, I think we will need to introduce similar mechanism for prevention of config reapplication for usb.json here: https://github.com/lf-edge/eve/blob/master/pkg/pillar/scripts/device-steps.sh#L311-L321
But this would be only needed for those case where USB ports are configured to be enabled even after onboarding.

Reply:
On the latter point, turns out that in /config/DevicePortConfig/ we process any *.json but from a USB stick we only process usb.json so I made the check look for a json suffix to accept both.

Reply:
Maybe I'm misreading it, but my understanding is:
- override.json (or any json file baked into the image under `/config/DevicePortConfig` is processed by NIM in `ingestDevicePortConfig` - this is taken care of by your code to avoid re-application
- usb.json (from USB with only DPC) is picked up by `device-steps.sh` and copied to `/run/global/DevicePortConfig`, so essentially directly to pubsub, where it is picked up by [this subscription](https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/nim/nim.go#L582-L592) and processed by [this handler](https://github.com/lf-edge/eve/blob/master/pkg/pillar/cmd/nim/nim.go#L720-L724). I don't think your PR will avoid re-application here, unless I missed something.  But I don't know if it is even important, because this DPC-only USBs are plugged into devices only temporarily. Also, after onboarding, USB ports are disabled by default so reapplication is no longer possible.

Reply:
Updated code in last commit to handle that path as well.